src/test/ut/ut_cortxfs_dir_ops.c

```diff
@@ -907,23 +908,36 @@ static void create_remove_subdir(void **state)
 		       pinode, "childdir");
 	ut_assert_int_equal(rc, 0);
 
-	memset(&after_rmdir, 0, sizeof(after_rmdir));
-	rc = cfs_getattr(ut_cfs_obj->cfs_fs, &ut_cfs_obj->cred,
-			 pinode, &after_rmdir);
+	rc = cfs_fh_from_ino(ut_cfs_obj->cfs_fs, pinode, &pfh_after_rmdir);
 	ut_assert_int_equal(rc, 0);
+	after_rmdir = cfs_fh_stat(pfh_after_rmdir);
 
 	/* if ctime after rmdir has not changed than assert */
-	if (after_rmdir.st_ctim.tv_sec == after_create.st_ctim.tv_sec &&
-	    after_rmdir.st_ctim.tv_nsec == after_create.st_ctim.tv_nsec) {
+	if (after_rmdir->st_ctim.tv_sec == after_create->st_ctim.tv_sec &&
+	    after_rmdir->st_ctim.tv_nsec == after_create->st_ctim.tv_nsec) {
 
 		ut_assert_true(0);
 	}
 	/* if mtime after rmdir has not changed than assert */
-	if (after_rmdir.st_mtim.tv_sec == after_create.st_mtim.tv_sec &&
-	    after_rmdir.st_mtim.tv_nsec == after_create.st_mtim.tv_nsec) {
+	if (after_rmdir->st_mtim.tv_sec == after_create->st_mtim.tv_sec &&
+	    after_rmdir->st_mtim.tv_nsec == after_create->st_mtim.tv_nsec) {
 
 		ut_assert_true(0);
 	}
+
+	if (pfh_before_create != NULL) {
+		cfs_fh_destroy(pfh_before_create);
+	}
+	if (pfh_after_create != NULL) {
+		cfs_fh_destroy(pfh_after_create);
+	}
+	if (pfh_after_rmdir != NULL) {
+		/* This test does not update parent FH after
```

Comment:
You should be using cfs_fh_destroy ad parent stat would have been to disk by cfs_rmdir. When cfs_rmdir or cfs_mkdir will work on FH and do all the updates in memroy then we have to dump those stat to disk explicity.

Reply:
Done!